examples/guided.py

```diff
@@ -352,7 +351,7 @@ def perform_installation(mountpoint):
 				installation.set_mirrors(archinstall.arguments['mirror-region'])  # Set the mirrors in the installation medium
 			if archinstall.arguments["bootloader"] == "grub-install" and has_uefi():
 				installation.add_additional_packages("grub")
-			installation.add_bootloader(archinstall.arguments["bootloader"])
+			installation.add_bootloader(archinstall.arguments["harddrive"], archinstall.arguments["bootloader"])
```

Comment:
The `archinstall.arguments["harddrive"]` argument should already be given to `installation` via: https://github.com/archlinux/archinstall/pull/595/files#diff-283d51110d004467e254215ad93e4dfcf0188beb22d9f878bb7088c12076dd2eR505

Is there a specific reason why we need to pass the harddrive argument once more?
I'm mainly asking because I'm re-working the logic for partitions and harddrive logics, and I'm wondering if I'll need to translate this over into my feature branch later on.

Reply:
> The `archinstall.arguments["harddrive"]` argument should already be given to `installation` via: https://github.com/archlinux/archinstall/pull/595/files#diff-283d51110d004467e254215ad93e4dfcf0188beb22d9f878bb7088c12076dd2eR505
> 
> Is there a specific reason why we need to pass the harddrive argument once more?
> I'm mainly asking because I'm re-working the logic for partitions and harddrive logics, and I'm wondering if I'll need to translate this over into my feature branch later on.

I was thinking this as well, but something weird that doesn't make this work with virtualbox. I tested with hard coded `/dev/sda` and it worked perfectly. So thus I simply pass the `archinstall.arguments["harddrive"]` to the function which holds the `/dev/sdX`. I have done this in a harry. I am sure the PR code need to be refactored. 

For this specific point, the only try worked correctly for me is:
`grub-install --target=i386-pc --recheck {_device.path}`

where `{_device.path}` = `/dev/sda`
`/boot` = `/dev/sda1`
`/` = `/dev/sda2`

Reply:
I think we should investigate why `/dev/sda` isn't detected why [real_device](https://github.com/archlinux/archinstall/blob/51eb05691a5845652a9dc583de05519f095961f3/archinstall/lib/installer.py#L491) isn't detected properly, if not we should fall back to [root_partition](https://github.com/archlinux/archinstall/blob/51eb05691a5845652a9dc583de05519f095961f3/archinstall/lib/installer.py#L491) instead of passing in more arguments.

This isn't easy to know for a new contributor, but we did extensive work to avoid passing in harddrive's as arguments to these functions because the installation should handle this within itself.